### DISCUSSION TITLE
Usage of Auditing for Deleted Models

### DISCUSSION THREAD

**Discussion by nilskretschmer** [owen-it/laravel-auditing#857] [Q&A] [answered]
Hi,

I am currently looking into this library for an ongoing project. After looking into the docs I am not 100% sure if it fits the project's needs.

Example:
I have some "Job" Model. This Job Model is associated with a user. If the user gets deleted, the job gets deleted (onDeleteCascade). How do I get audits from a deleted model? Are audits lost if a model is deleted? Do I need to use softDelete on all models I want to have audits for?

Because the Relationship between the auditable morph table and the deleted model is lost I think the only way would be to query the audit table itself. But I am not sure if the Audits itself are also cascade-deleted if their related model was deleted.

Can someone help me out?

**Comment by parallels999** [comment]
> I am not sure if the Audits itself are also cascade-deleted

Not, it don't

>Are audits lost if a model is deleted?

No

>I need to use softDelete on all models I want to have audits for?

The ideal is to use it, so you can use `restore`, or revert changes

**Reply by parallels999** [reply, marked as answer]
> How could I get all audits for my (lets stay in the example) Job model - including those of deleted jobs?

From db with `Audit` model, use a where for `model_type='App\Models\Jobs'`, still laravel eloquent , you can do whatever you want

**Comment by nilskretschmer** [comment]
@parallels999 
Thank you for clarifying this. In my case I won't need restore/revert changes functionality. So I won't use SoftDeletes (for now).

How could I get all audits for my (lets stay in the example) ```Job``` model - including those of deleted ```jobs```?
